### DISCUSSION TITLE
大批量jar包生成facts的内存占用问题

### DISCUSSION THREAD

**Discussion by Y4er** [BytecodeDL/ByteCodeDL#4] [General]
在我有几百个jar包使用https://github.com/BytecodeDL/soot-fact-generator 生成fact时会不会产生oom的异常？

之前用tabby会产生这个问题，个人用磁盘占用替代内存占用解决了，但是这个项目不太熟悉，先问一下= =

如果有oom异常的问题，那么除了加内存以外有没有其他的解决办法？

**Comment by waderwu** [comment]
生成fact可以像你说的，可以边解析，边存到磁盘，但是如果项目过大，我觉得souffle在分析的时候可能也会有问题

**Reply by waderwu** [reply]
今天翻源码，还看到了`--lowMem`这个参数，这个参数有助于减少内存的使用
